app/components/Form.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import { StyleSheet, Text, View, TextInput, TouchableOpacity, AsyncStorage} from 'react-native';
+import { StackNavigator } from 'react-navigation';
+
+export default class Form extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state={
+            name: '',
+            email: '',
+            emailValidation : true,
+            password: '',
+            passwordValidate: true,
+            showError: false,
+            user: '',
+            errorMsg: '',
+        }
+        this.verifyUser = this.verifyUser.bind(this);
+        this.saveUser = this.saveUser.bind(this);
+        this.showErrorMessage = this.showErrorMessage.bind(this);
+        this.saveLogState = this.saveLogState.bind(this);
+    }
+
+    validate(text,type){
+        var email= /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+        var num =/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/;
+        if(type=="email"){
+            if (email.test(text)) {
+                this.setState({ emailValidation: true, email: text });
+            } else {
+                this.setState({ emailValidation: false });
+            }
+        }else if(type=="password"){
+            if (num.test(text)) {
+                this.setState({ passwordValidate: true, password: text,showError: false});
+            }else{
+                if (this.props.type == 'Signup') {
+                    let msg = 'password must contain at least: 1 uppercase character, 1 number ,must be 8 characters or longer';
+                    this.showErrorMessage(msg, true);
+                }
+                this.setState({ passwordValidate: false });
+            }
+        }else if(type == "name"){
+            this.setState({ name : text})
+        }
+    }
+
+    async verifyUser(key){
+        try {
+            await AsyncStorage.getItem(key).then((val) => {
+                this.setState({ user: JSON.parse(val)});
+                });
+        } catch (error) {
+            console.log('Error: ', error);
+        }
+        
+    }
+
+    
+    async saveUser (data,key) {
+        try {
+            let user = JSON.stringify(data);
+            
+            await AsyncStorage.setItem(key, user);
+        } catch (error) {
+            // Error saving data
+            console.log(error);
+            
+        }
+    };
+
+    showErrorMessage(msg,state){
+        this.setState({
+            errorMsg: msg,
+            showError: state
+        });
+    }
+
+    saveLogState(){        
+        this.props.screenProps.isLoggedIn();
+    }
+
+    buttonHandler(){
+
+        let {type} = this.props;
+        let { name, email, password } = this.state;
+        
+        if (type =='Login'){
```

Comment:
you can refactor those multiple conditional in one `if(type === 'Login' && email && password)` two things here 1. '' === false, so when you are validating a empty string you don't need to do if(email !== ''), you can do if(email) and this would work,  2. always use === and !== , remember, == and !== compare value and ===  and !== compare type and value.